### DISCUSSION TITLE
为什么流媒体直播的延迟很高 · Why's THE Design? · /whys-the-design-live-streaming-latency

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#594] [Announcements]
https://draveness.me/whys-the-design-live-streaming-latency 

 通信技术的发展促进了视频点播和直播业务的兴起，4G 和 5G 网络技术的进步也使得流媒体技术变得越来越重要，本文不会介绍网络对直播业务的影响，而是会分析一个直播中常见的现象 — 主播和观众之间能够感觉到的明显网络延迟。除了业务上要求的延迟直播之外，有哪些因素会导致视频直播的延迟这么高呢？

**Comment by bloodyparadise** [comment]
“假设我们需要看一部时长为 2 小时的 1080p、60FPS 的电影，如果每个像素需要 2 字节的存储，那么整部电影需要占用如下所示的资源：”
这行下边是不是缺了幅图片？

**Comment by draveness** [comment]
> “假设我们需要看一部时长为 2 小时的 1080p、60FPS 的电影，如果每个像素需要 2 字节的存储，那么整部电影需要占用如下所示的资源：”
> 这行下边是不是缺了幅图片？

公式竟然被吃了...

**Comment by hzmrwu** [comment]
HLS的额外开销应该是m3u8描述文件本身？

**Comment by draveness** [comment]
> @hzmrwu
> HLS的额外开销应该是m3u8描述文件本身？

不好意思，没明白你的问题

**Comment by gangbanlau** [comment]
不妨看看 HLS 和 DASH 低延时的现状。利用块传输的方式(chunked transfer) 或者 Apple LL-HLS，GOP 的长度不再和时延挂钩。在浏览器下也可以实现低延时。CDN 缓存解决不了最后一公里的网络抖动，这个通常靠 ABR 去解决

**Comment by draveness** [comment]
> 不妨看看 HLS 和 DASH 低延时的现状。利用块传输的方式(chunked transfer) 或者 Apple LL-HLS，GOP 的长度不再和时延挂钩。在浏览器下也可以实现低延时。CDN 缓存解决不了最后一公里的网络抖动，这个通常靠 ABR 去解决

DASH 确实是个解决方案，我们也在看这个东西，不过国内 CDN 支持比较差

**Comment by gangbanlau** [comment]
DASH 运维要求高，技术上似乎没有必要因为延时原因从 HLS 转 DASH。请问国内CDN 不支持 http 1.1 chunked transfer吗?

**Comment by draveness** [comment]
> DASH 运维要求高，技术上似乎没有必要因为延时原因从 HLS 转 DASH。请问国内CDN 不支持 http 1.1 chunked transfer吗?

不太清楚，用得比较多就是 HTTP-FLV 和 HLS

**Comment by gangbanlau** [comment]
> > DASH 运维要求高，技术上似乎没有必要因为延时原因从 HLS 转 DASH。请问国内CDN 不支持 http 1.1 chunked transfer吗?
> 
> 不太清楚，用得比较多就是 HTTP-FLV 和 HLS

ok. now HLS and DASH can target 3s glass to glass latency in production in scale. I hope this is helpful for you. 
sorry don't have Chinese input right now.

**Comment by draveness** [comment]
> > > DASH 运维要求高，技术上似乎没有必要因为延时原因从 HLS 转 DASH。请问国内CDN 不支持 http 1.1 chunked transfer吗?
> > 
> > 
> > 不太清楚，用得比较多就是 HTTP-FLV 和 HLS
> 
> ok. now HLS and DASH can target 3s glass to glass latency in production in scale. I hope this is helpful for you.
> sorry don't have Chinese input right now.

Thanks for the input, and we planned to used DASH as the streaming protocol oversea.

**Comment by zinwalin** [comment]
> @draveness
> > > > DASH 运维要求高，技术上似乎没有必要因为延时原因从 HLS 转 DASH。请问国内CDN 不支持 http 1.1 chunked transfer吗?
> > > 
> > > 
> > > 不太清楚，用得比较多就是 HTTP-FLV 和 HLS
> > 
> > ok. now HLS and DASH can target 3s glass to glass latency in production in scale. I hope this is helpful for you.
> > sorry don't have Chinese input right now.
> 
> Thanks for the input, and we planned to used DASH as the streaming protocol oversea.

xiu de qi fei. :)

**Comment by wujunze** [comment]
HLS  兼容性平台应该是最好的了  延迟高
RTMP 延迟低  H5不能直接播放

**Comment by gangbanlau** [comment]
Apple LL-HLS 可以在5秒延时提供高质量, 已经接近商用

**Comment by MenThu** [comment]
“与微信视频等 1 对 1 的实时通信相比，视频流的生产方和消费方之间的链路极长”
直播的链路比实时音视频的链路长在哪呢？

**Comment by gangbanlau** [comment]
> “与微信视频等 1 对 1 的实时通信相比，视频流的生产方和消费方之间的链路极长”
> 直播的链路比实时音视频的链路长在哪呢？


商业直播的环节多。低质量的个人直播没有那些硬性的限制，利用实时通信技术实现小规模直播也是有不少公司在做的。HLS/DASH更多的考量是一个生态圈，极低成本，高质量，和优秀的客户体验。

**Comment by jeepli** [comment]
高实时性要求的直播采用RTC方式，一般不采用httpflv + cdn方式

**Comment by gama-ray** [comment]
大佬，能讲一讲现在的互联网直播业务会用到IP组播吗
